File: python/docs/examples/_async/mobile-use/mobile_beta_screenshot.py
import asyncio
import os

from agentbay import AsyncAgentBay, CreateSessionParams
from agentbay import AgentBayError


async def _set_low_android_resolution(session) -> None:
    cmds = [
        "wm size 720x1280",
        "wm density 160",
    ]
    for cmd in cmds:
        r = await session.command.execute_command(cmd)
        if not r.success:
            raise RuntimeError(f"Command failed: {cmd} error={r.error_message}")


async def main() -> None:
    api_key = os.environ.get("AGENTBAY_API_KEY")
    if not api_key:
        raise RuntimeError("AGENTBAY_API_KEY is not set")

    agent_bay = AsyncAgentBay(api_key=api_key)
    session = None
    try:
        result = await agent_bay.create(CreateSessionParams(image_id="imgc-0ab5takhnmlvhx9gp"))
        if not result.success or not result.session:
            raise RuntimeError(f"Failed to create session: {result.error_message}")
        session = result.session

        await _set_low_android_resolution(session)

        os.makedirs("./tmp", exist_ok=True)

        start = await session.mobile.start_app("monkey -p com.android.settings 1")
        if not start.success:
            raise RuntimeError(f"Failed to start Settings: {start.error_message}")
        await asyncio.sleep(2)

        s1 = await session.mobile.beta_take_screenshot()
        with open("./tmp/mobile_beta_screenshot.png", "wb") as f:
            f.write(s1.data)
        print(f"Saved ./tmp/mobile_beta_screenshot.png ({len(s1.data)} bytes)")

        try:
            s2 = await session.mobile.beta_take_long_screenshot(max_screens=2, format="png")
            with open("./tmp/mobile_beta_long_screenshot.png", "wb") as f:
                f.write(s2.data)
            print(f"Saved ./tmp/mobile_beta_long_screenshot.png ({len(s2.data)} bytes)")
        except AgentBayError as e:
            print(f"Long screenshot failed: {e}")
    finally:
        if session is not None:
            await session.delete()


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/docs/examples/mobile-use/mobile-beta-screenshot.ts
import { AgentBay } from "wuying-agentbay-sdk";
import fs from "node:fs";
import path from "node:path";

async function main() {
  const apiKey = process.env.AGENTBAY_API_KEY;
  if (!apiKey) {
    throw new Error("AGENTBAY_API_KEY is not set");
  }

  const agentBay = new AgentBay({ apiKey });
  const create = await agentBay.create({ imageId: "imgc-0ab5takhnmlvhx9gp" });
  if (!create.success || !create.session) {
    throw new Error(`Failed to create session: ${create.errorMessage || ""}`);
  }

  const session = create.session;
  try {
    await new Promise((resolve) => setTimeout(resolve, 15000));

    const cmds = ["wm size 720x1280", "wm density 160"];
    for (const c of cmds) {
      const r = await session.command.executeCommand(c, 10000);
      if (!r.success) {
        throw new Error(`Command failed: ${c} error=${r.errorMessage || ""}`);
      }
    }

    const start = await session.mobile.startApp("monkey -p com.android.settings 1");
    if (!start.success) {
      throw new Error(`Failed to start Settings: ${start.errorMessage || ""}`);
    }
    await new Promise((resolve) => setTimeout(resolve, 2000));

    const nav = await session.command.executeCommand("am start -a android.settings.SETTINGS", 10000);
    if (!nav.success) {
      throw new Error(`Command failed: am start -a android.settings.SETTINGS error=${nav.errorMessage || ""}`);
    }
    await new Promise((resolve) => setTimeout(resolve, 2000));

    const outDir = "./tmp";
    fs.mkdirSync(outDir, { recursive: true });

    const s1 = await session.mobile.betaTakeScreenshot();
    if (!s1.success) {
      throw new Error(`beta_take_screenshot failed: ${s1.errorMessage || ""}`);
    }
    const p1 = path.join(outDir, "mobile_beta_screenshot.png");
    fs.writeFileSync(p1, Buffer.from(s1.data));
    console.log(
      `Saved ${p1} (${s1.data.length} bytes, size=${s1.width}x${s1.height})`
    );

    const s2 = await session.mobile.betaTakeLongScreenshot(2, "png");
    if (!s2.success) {
      console.error(`beta_take_long_screenshot failed: ${s2.errorMessage || ""}`);
      return;
    }
    const p2 = path.join(outDir, "mobile_beta_long_screenshot.png");
    fs.writeFileSync(p2, Buffer.from(s2.data));
    console.log(
      `Saved ${p2} (${s2.data.length} bytes, size=${s2.width}x${s2.height})`
    );
  } finally {
    await session.delete();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

